core/src/main/java/org/opensearch/sql/ast/expression/Field.java

```diff
@@ -51,4 +51,24 @@ public List<UnresolvedExpression> getChild() {
   public <R, C> R accept(AbstractNodeVisitor<R, C> nodeVisitor, C context) {
     return nodeVisitor.visitField(this, context);
   }
+
+  public boolean isNested() {
+    return this.toString().contains(".");
+  }
```

Comment:
I'm wondering, rather than check in this tricky way as what legacy does, should we add nested data type? Because in V2 engine, we have strict semantic check mechanism and can define function signature, ex. `nested(nestDataType)`.

Reply:
We have added stricter parser grammar for the nested function. We could extend the `Field` class for nested types, or add some functionality through optional arguments in a new `Field` constructor. Are either of these options what you had in mind?